### PR TITLE
Fix breaking the font inspector

### PR DESCRIPTION
**Pull request by Connum**
The font inspector threw an error and showed "Undefined" for the naming table.

### CODE REVIEW

File: font-inspector.html [hunk 1]
```diff
@@ -161,22 +163,32 @@ <h1>Free Software</h1>
...
-        var langs = sortKeys(translations);
-        for (var j = 0; j < langs.length; j++) {
-            var lang = langs[j];
-            var esclang = escapeHtml(lang);
-            html += '<span class="langtag">' + esclang
-                + '</span> <span class="langname" lang=' + esclang + '>'
-                + escapeHtml(translations[lang]) + '</span> ';
+    let html = '';
+    ['unicode', 'macintosh', 'windows'].forEach( platform => {
+        platformNames = names?.[platform];
```
Comment (ILOVEPIE) [inline font-inspector.html]: where is the let declaration, we don't want these variables to be global.

File: font-inspector.html [hunk 2]
```diff
@@ -161,22 +163,32 @@ <h1>Free Software</h1>
...
+
+        if ( ! platformNames ) return;
+
+        html += `<h4>${platform}</h4><dl id="name-table-${platform}">`;
+
+        properties = sortKeys(platformNames);
```
Comment (ILOVEPIE) [inline font-inspector.html]: see above.

File: font-inspector.html [hunk 3]
```diff
@@ -161,22 +163,32 @@ <h1>Free Software</h1>
...
-            html += '<span class="langtag">' + esclang
-                + '</span> <span class="langname" lang=' + esclang + '>'
-                + escapeHtml(translations[lang]) + '</span> ';
+    let html = '';
+    ['unicode', 'macintosh', 'windows'].forEach( platform => {
+        platformNames = names?.[platform];
+
+        if ( ! platformNames ) return;
+
+        html += `<h4>${platform}</h4><dl id="name-table-${platform}">`;
```
Comment (ILOVEPIE) [inline font-inspector.html:172]: see above.

**Review by ILOVEPIE — CHANGES_REQUESTED** [review]
we should probably make sure we're not polluting the global scope.

File: font-inspector.html (hunk 3 above)
Comment (ILOVEPIE) [reply font-inspector.html:172]: nvmd this one's fine.

File: font-inspector.html (hunk 1 above)
Comment (Connum) [reply font-inspector.html]: Good catch, and it shows why code review is so important! That's what happens when you frequently switch between JS and PHP development... 😉

**Comment by Connum** [comment]
fixed the variable scoping issue, ready for review

File: font-inspector.html [hunk 4]
```diff
@@ -161,22 +163,32 @@ <h1>Free Software</h1>
...
-        html += '<dt>'+escapeHtml(property)+'</dt><dd>';
-        var translations = names[property];
-        var langs = sortKeys(translations);
-        for (var j = 0; j < langs.length; j++) {
-            var lang = langs[j];
-            var esclang = escapeHtml(lang);
-            html += '<span class="langtag">' + esclang
-                + '</span> <span class="langname" lang=' + esclang + '>'
-                + escapeHtml(translations[lang]) + '</span> ';
+    let html = '';
```
Comment (ILOVEPIE) [inline font-inspector.html]: Actually, the rest of this is using var which means older browsers (like old iphones etc) will work with it, which do we want to use? We can fix this with a later commit though.

**Review by ILOVEPIE — APPROVED** [review]
We might want to address the one comment, or I can address it in a later commit.

File: font-inspector.html (hunk 4 above)
Comment (ILOVEPIE) [reply font-inspector.html]: @Connum see above.

Comment (Connum) [reply font-inspector.html]: Yeah, you're right, let's not break backwards compatibility where we don't need to.

File: font-inspector.html [hunk 5]
```diff
@@ -162,21 +164,31 @@ <h1>Free Software</h1>
...
-        var translations = names[property];
-        var langs = sortKeys(translations);
-        for (var j = 0; j < langs.length; j++) {
-            var lang = langs[j];
-            var esclang = escapeHtml(lang);
-            html += '<span class="langtag">' + esclang
-                + '</span> <span class="langname" lang=' + esclang + '>'
-                + escapeHtml(translations[lang]) + '</span> ';
+    ['unicode', 'macintosh', 'windows'].forEach( platform => {
+        let platformNames = names?.[platform];
```
Comment (ILOVEPIE) [inline font-inspector.html]: This still has let as well as optional chaining, should probably be:
```js
var platformNames = (names?names[platform]:null);
```

**Review by ILOVEPIE — CHANGES_REQUESTED** [review]
@Connum one last change required, you may just want to amend the last commit as you missed one let.

File: font-inspector.html (hunk 5 above)
Comment (ILOVEPIE) [reply font-inspector.html]: fixed the comment, there was a misunderstanding.

Comment (Connum) [reply font-inspector.html]: Couldn't amend because I'm currently on mobile and limited to the web editor

**Review by ILOVEPIE — APPROVED** [review]
LGTM

**Comment by ILOVEPIE** [comment]
@Connum Btw, based on the kinds of projects I see in your github, I think you'd like the project that I've been working on that depends on opentype.js. It's called [SABRE.js](https://github.com/SABRE-JS/SABRE.js) which stands for Substation Alpha suBtitles REnderer, It's actually pretty close to libass in terms of rendering output at this point.

**Comment by Connum** [comment]
> @Connum Btw, based on the kinds of projects I see in your github, I think you'd like the project that I've been working on that depends on opentype.js. It's called [SABRE.js](https://github.com/SABRE-JS/SABRE.js) which stands for Substation Alpha suBtitles REnderer, It's actually pretty close to libass in terms of rendering output at this point.

Yeah, I already looked at it! 😃

**Pull request merged**